### PR TITLE
HPCC-19120 Internal errors from fileview code on indexes with payload

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Manual testing in eclwatch

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

**Comment by richardkchapman** [comment]
Those regression suite changes may actually be correct - will look more closely

File: ecl/hql/hqlutil.cpp [hunk 1]
```diff
@@ -9854,9 +9854,13 @@ static IFieldFilter * createIfBlockFilter(IRtlFieldTypeDeserializer &deserialize
 unsigned buildRtlRecordFields(IRtlFieldTypeDeserializer &deserializer, unsigned &idx, const RtlFieldInfo * * fieldsArray, IHqlExpression *record, IHqlExpression *rowRecord)
 {
     unsigned typeFlags = 0;
+    unsigned numFields = getFlatFieldCount(record);
+    unsigned payloadFields = numPayloadFields(record);
     ForEachChild(i, record)
     {
         unsigned fieldFlags = 0;
+        if (idx >= numFields-payloadFields)
```
Comment (ghalliday) [inline ecl/hql/hqlutil.cpp]: I'm not sure that this idx test is correct if it is derived from getFlatFieldCount - if the index record happened to extend another index.  (very weird code).  firstPayloadField() may be safer.

**Comment by ghalliday** [comment]
@richardkchapman you were going to look at the regression suite changes.

**Comment by richardkchapman** [comment]
I don't think the regression suite differences should have happened - I need to re-investigate (if I can remember how this was supposed to work).

**Comment by richardkchapman** [comment]
@ghalliday Please re-review

**Comment by ghalliday** [comment]
@richardkchapman why is it failing those regression tests?

**Comment by richardkchapman** [comment]
Because it is setting the payload bit on all fields in a record.
Not sure why though - numPayload should be 0 in these cases

**Comment by richardkchapman** [comment]
@ghalliday

**Pull request merged**